src/translator/text_processor.cpp

```diff
@@ -138,5 +138,39 @@ void TextProcessor::wrap(Segment &segment, std::vector<string_view> &wordRanges,
   }
 }
 
+void TextProcessor::processFromAnnotation(AnnotatedText &source, Segments &segments) const {
+  std::string copySource = source.text;
+  AnnotatedText replacement(std::move(copySource));
+
+  for (size_t s = 0; s < source.numSentences(); s++) {
+    // This is our sentenceStream
+    ByteRange sentenceByteRange = source.sentenceAsByteRange(s);
+
+    // Fool tokenization using ByteRanges into looking at replacement. They're same, so okay.
+    marian::string_view sentence{&replacement.text[sentenceByteRange.begin], sentenceByteRange.size()};
+
+    std::vector<string_view> wordRanges;
+    Segment segment = tokenize(sentence, wordRanges);
```

Comment:
Yes, there was some talk in a meeting about controlling the growth of memory. Given we're working around so many things to provide pivoting - different vocabs, no way of any control over tokens - it made sense to simply relax it for pivoting.

Not that the previous control on the non-pivot path still exists by means of truncation in the text processor. It's just the generated text (pivot in-betweens) that we have no control over. The trouble arises due to constraints on alignments and mappings which is required for features like HTML translation.

It might be possible to constrain sequence lengths by cutting and merging - introducing notions of subsentences, but it's too much work and will further delay in this PR.